src/main/java/com/salesforce/mirus/HerderStatusMonitor.java

```diff
@@ -91,42 +90,50 @@ private void onConnectors(Throwable error, Collection<String> connectorNames) {
 
   private void processConnector(String connectorName) {
     ConnectorStateInfo stateInfo = herder.connectorStatus(connectorName);
-    if (workerId.equals(stateInfo.connector().workerId())) {
-      if (stateInfo.connector().state().equals(ConnectorStatus.State.FAILED.toString())) {
-        connectorJmxReport.incrementConnectorRestartAttempts(connectorName);
-        if (autoRestartConnectorEnabled) {
+
+    if (autoRestartConnectorEnabled) {
```

Comment:
@pdavidson100  what do you think of publishing the JMX metrics first then do the restart attempt? I believe the state will flip to something transient like `UNASSIGNED` or something before before it is `RUNNING`. I believe reporting `FAILED` and then attempting the restart will make more sense.